### DISCUSSION TITLE
Method for getting an IEDs dial code

### DISCUSSION THREAD

**Discussion by The-Pink-Panther** [acemod/ACE3#8323] [Q&A] [answered]
I am currently making a mission in which I want an AI IED spotter to be able to detonate a cellphone IED after seing a player. I am currently using this script to spawn the IED:
`iedOne = [player, [2576.687,8859.097,0], 0, "IEDLandBig_Remote_Mag", "Cellphone", [player, this], placeHolder] call ace_explosives_fnc_placeExplosive; ` 
This works as intended and adds the IED to the speed dial menu, the problem comes when trying to detonate the device. I'm trying to use fnc_dialPhone.sqf but this requires the dial code, fnc_getSpeedDialExplosive.sqf returns the IED from the dial code, a function that did the opposite and returned the dial code for a specified IED would be great.

**Comment by Whigital** [comment, marked as answer]
Just scanned through the code, and Cellphone connected IEDs are added into an array called `ace_explosives_CellphoneIEDs`, which is also broadcasted to all clients.

This array contains the ied, code and trigger delay:
https://github.com/acemod/ACE3/blob/af66a9881c6a49ff4aa8dc1c78ef70c9048962a5/addons/explosives/functions/fnc_addCellphoneIED.sqf#L47-L49

So if you have a reference to the IED object, it _should_ just be a simple matter of extracting it from the array:
```sqf
params [["_ied", objNull]];

if (isNull _ied) exitWith {
    diag_log "Null object passed to get IED code ....";
    -1
};

private _iedIdx = (ace_explosives_CellphoneIEDs findIf {(_x # 0) == _ied});

if (_iedIdx == -1) exitWith {
    diag_log "IED not found in Cellphone array ....";
    -1
};

private _iedData = (ace_explosives_CellphoneIEDs # _iedIdx);

(_iedData # 1)
```

**Reply by The-Pink-Panther** [reply]
Checked and it works, thanks.
